src/libappimage/core/impl/StreambufType1.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// system
+#include <vector>
+#include <streambuf>
+
+// libraries
+#include <archive.h>
+
+namespace appimage {
+    namespace core {
+        namespace impl {
+            /**
+             * Provides a streambuf implementation for reading type 1 AppImages
+             * by means of libarchive.
+             *
+             * For more details about streambuf see https://gcc.gnu.org/onlinedocs/libstdc++/manual/streambufs.html
+             */
+            class StreambufType1 : public std::streambuf {
+            public:
+                /**
+                 * Create an streambuf_type_1 object from an archive <a> pointer
+                 * with a buffer size of <size>
+                 * @param a opened archive struct from libarchive
+                 * @param size buffer size
+                 */
+                StreambufType1(archive* a, unsigned long size);
+
+                // Creating copies of this object is not allowed
+                StreambufType1(StreambufType1& other) = delete;
```

Comment:
There's more method to take care in C++ >=11 (move operators, rule of 5).

Reply:
Added.